modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+            doThrow(new NotFoundException()).when(userManager).deleteUser(id);
+            SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        }
+    }
+
+    @DataProvider(name = "dataForTestDeleteUserFailsINCharonExceptionOnly")
+    public Object[][] dataToTestDeleteUserCharonExceptionOnly()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        return new Object[][]{
+                {id, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteUserFailsINCharonExceptionOnly")
+    public void testDeleteUserCharonExceptionOnly(String id, int expectedScimResponseStatus)
+            throws NotFoundException, NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/" + USER_ID);
+
+        doThrow(new CharonException()).when(userManager).deleteUser(id);
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+        SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestUpdateWithPUTSuccess")
+    public Object[][] dataToTestUpdateWithPUTSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
 
         SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
         JSONDecoder decoder = new JSONDecoder();
-        return decoder.decodeResource(scimObjectString, schema, new User());
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING, schema, new User());
+        String id = userOld.getId();
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_UPDATE, schema, new User());
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, "userName", null,
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, "userName", "emails",
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, null, "emails",
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, null, null,
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
```

Comment:
include data sets with empty strings for `attributes`  and `excludeAttributes`

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11

Reply:
is this fixed?

Reply:
Fixed in  commit af77bf94a86e4b649d26ce314d5408b2fde6750d